app/jobs/hackathons/send_digests_job.rb

```diff
@@ -0,0 +1,27 @@
+class Hackathons::SendDigestsJob < ApplicationJob
+  sidekiq_options retry: 20 # up to 6.5 days
+
+  def perform
+    sent_digests = []
+    current_subscribers.find_each do |subscriber|
+      next unless new_digest_pertinent?(subscriber)
+
+      digest = subscriber.digests.new
+
+      digest.save! unless digest.listings.none?
+      sent_digests << digest if digest.persisted?
+    end
+  ensure
+    Hackathons::DigestMailer.admin_summary(sent_digests).deliver_later if sent_digests.any?
+  end
+
+  private
+
+  def current_subscribers
+    User.includes(:subscriptions).where(subscriptions: {status: :active})
+  end
+
+  def new_digest_pertinent?(subscriber)
+    subscriber.digests.last&.created_at&.before? 6.days.ago
```

Comment:
This would be `nil`(false) if there no last digest. I'd suggest inverting the statement (like what it was previously) or add a guard clause here.

```ruby
# I prefer this solution
next if has_recent_digest?(subscriber)

def has_recent_digest?(subscriber)
  subscriber.digests.last&.created_at&.after? 6.days.ago
end
```

or

```ruby
def new_digest_pertinent?(subscriber)
  return true if subscriber.digests.empty?

  subscriber.digests.last.created_at.before? 6.days.ago
end
```